backend/infrahub/graphql/auth/query_permission_checker/default_branch_checker.py

```diff
@@ -38,6 +25,14 @@ async def check(
         query_parameters: GraphqlParams,
         branch: Branch,
     ) -> CheckerResolution:
+        can_edit_default_branch = False
+        for permission_backend in registry.permission_backends:
+            can_edit_default_branch = await permission_backend.has_permission(
+                db=db, account_id=account_session.account_id, permission=self.permission_required, branch=branch
+            )
+            if can_edit_default_branch:
+                break
```

Comment:
We might potentially need to revisit this if we add a "deny" permission to the mix and we want to see if this global permission is denied anywhere.

Reply:
Absolutely yes.